### DISCUSSION TITLE
Help - recording UE transmit power and amount of UL control data exchanged

### DISCUSSION THREAD

**Discussion by victorfariasm** [srsran/srsRAN_Project#607] [General Help]
Hello everyone, 

I've been trying to compare 4G and 5G in terms of UE power consumption.  
More specifically, I have a feeling that 5G consumes more UE power, even though 5G is supposed to be more energy efficient.

In order to check this hypothesis, my first attempt was to record UE transmit power and the amount of control data exchanged (bits/msgs/etc.) between UE and eNB/gNB and then correlate these data.
However, so far, I've not yet succeeded to find and record these metrics.
Do you have any suggestion on how I can record them?
Or, do you have any other suggestion of other metrics that I should record (and how to do this) in order to check if my hypothesis is correct or not?

Brs,
Victor.

**Comment by pgawlowicz** [comment]
Hi Victor,

The easiest option is to take the `gnb.log` file and look for lines with `PUSCH:` and `PUCCH:`. It shows the UL transmission from UEs ( contains information on Transport Block Size, SNR, etc). But it only covers one part of what you need.

If you want to collect the information on the UE side, you might try using the `Network Signal Guru` app. It shows a lot of information if you have a rooted phone. 

Note that the power savings in 5G come from the power-saving mechanisms (like DRX, BWP switching, reducing PDCCH monitoring, etc). I doubt it will use less power during the data transmission. On the other side, the power consumption is specific to chip design/implementation, so hard to compare. 

Best,
Piotr
